index.js

```diff
@@ -177,3 +221,45 @@ function findBestSubtitleFile(list) {
     .filter((x) => x.subs)
     .sort((x, y) => x.priority > y.priority)[0];
 }
+
+function processFormats(formats) {
+  const filteredFormats = formats.filter(({ format_id, fps, height, vcodec }) => format_id !== 'source' && !format_id.startsWith('http') && vcodec && vcodec !== 'none' && !vcodec.includes('av01') && !vcodec.includes('vp9') && !vcodec.includes('vp09') && ((height >= 1080 && fps <= 30) || (height < 1080)))
+  .sort((prevFormat, nextFormat) => nextFormat.tbr - prevFormat.tbr);
+
+  const selected4K = filteredFormats.find((format) => (format.height === 2160 && format.acodec !== 'none')) || filteredFormats.find((format) => format.height === 2160);
+  const selectedHd = filteredFormats.find((format) => (format.height === 1080 && format.acodec !== 'none')) || filteredFormats.find((format) => format.height === 1080);
+  const selectedLowQuality = filteredFormats.find((format) => (format.height === 720 && format.acodec !== 'none')) || filteredFormats.find((format) => format.height === 720) || filteredFormats.find((format) => (format.height <= 720 && format.acodec !== 'none')) || filteredFormats.find((format) => (format.height <= 720));
```

Comment:
could this just be

`const selectedLowQuality = filteredFormats.find((format) => format.height <= 720))`

?

Reply:
Nevermind this comment. I guess you are prioritizing the 720p combined file, then a 720p video, then a video smaller than 720p.

Reply:
Yeah exactly.